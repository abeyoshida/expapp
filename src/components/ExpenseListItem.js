import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  //const createdAtSafe = (moment.isMoment(createdAt)) ? createdAt.format("dddd, MMMM Do YYYY, h:mm:ss a") : createdAt;
  const createdAtSafe = '';
  return (
      <div>
        <Link to={`/edit/${id}`}>
          <h3>{description}</h3>
        </Link>
        {/*<p>{amount} - {createdAt}</p>*/}
        <p>{amount} - {createdAtSafe}</p>
      </div>
    );
}
export default ExpenseListItem;
