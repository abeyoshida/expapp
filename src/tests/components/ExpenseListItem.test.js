import React from 'react';
import { shallow } from 'enzyme';
import expensesDummy from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('render an ExpenseListItem from expensesDummy', () => {
    const wrapper = shallow(<ExpenseListItem key={expensesDummy[2].id} expenses={ expensesDummy[2] }/>);
    expect(wrapper).toMatchSnapshot();
});
