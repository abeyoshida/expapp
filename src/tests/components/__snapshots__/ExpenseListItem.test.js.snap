// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render an ExpenseListItem from expensesDummy 1`] = `
<[object Object]
  className="list-item"
  to="/edit/undefined"
>
  <div>
    <h3
      className="list-item__title"
    />
    <span
      className="list-item__sub-title"
    >
      December 31st 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $0.00
  </h3>
</[object Object]>
`;
